### DISCUSSION TITLE
_Impl.h

### DISCUSSION THREAD

**Discussion by Hsubtnarg** [khoih-prog/Ethernet_Generic#9] [General]
Thanks for your code, it is great. I have being trying to get something like this for ages. But why are all your code files _Impl.h instead of .cpp? It seems odd to have all your code in header files and makes compiling slow.

Thanks again

**Comment by khoih-prog** [comment]
> makes compiling slow

That's the small trade-off to have better code using h-only style, which I have been used so far and won't come back to regular `cpp` style. You also possibly need to upgrade your computer (CPU, speed, memory, etc.) as I haven't seen the `slowing` effect yet, even large projects.

I'll let you do some research yourself as well as trying to convert to `cpp` and see the bad effects to the code.
